### ISSUE TITLE
Bug: DeprecatedKoinApi false positive on viewModel with named dependencies

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#13]
## Summary

The `DeprecatedKoinApi` rule suggests migrating from `viewModel { }` to `viewModelOf()`, but `viewModelOf()` (and other `*Of` DSL variants like `singleOf`/`factoryOf`) **cannot pass named/qualified dependencies**. This makes the suggestion invalid for a common use case.

## Current Behavior

```kotlin
val appModule = module {
    viewModel { MyViewModel(get(), get(named("api"))) } // ← flagged
}
```

**Detekt output:**
```
DeprecatedKoinApi - Deprecated API 'viewModel' used; migrate to 'viewModelOf()'
```

## Expected Behavior

The rule should **not** flag `viewModel { }` when the lambda body contains calls with `named()` or `qualifier()` parameters, since `viewModelOf()` cannot handle this case.

As noted in the [Koin documentation](https://insert-koin.io/docs/reference/koin-android/get-instances/#dsl):
> NOTE: viewModelOf() doesn't support named parameters

Related Koin issue: https://github.com/InsertKoinIO/koin/issues/1421#issuecomment-1422150092

## Proposed Fix

Before suggesting migration to `*Of()` variants, check if the lambda body uses:
- `get(named(...))` or `get(qualifier(...))`
- `inject(named(...))` or `inject(qualifier(...))`

If qualified dependencies are used, skip the report.

## Impact

6 false positives in a real project that uses named dependencies with viewModel.

## Environment

- Koin-Detekt 0.4.0
- Detekt 1.23.8
